### PR TITLE
add ldap idp test + few fixes

### PR DESCRIPTION
**Pull request by shellymiron**
Added case [OCP-63332](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-63332) for LDAP idp.
added few fixes to our current IDP implementation. 

image

### CODE REVIEW

File: tests/utils/constants/constants.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ var (
 	WorkSpace           = "WORKSPACE"
 	RHCSPrefix          = "rhcs"
 	TFYAMLProfile       = "tf_cluster_profile.yml"
+	LdapURL             = "ldap://ldap.forumsys.com/dc=example,dc=com?uid"
```
Comment (tshefi) [inline tests/utils/constants/constants.go:36]: This url points to an external LDAP resource, for manual testing every once in a while I assume this should be fine. For automation we should probably find/create an internal LDAP resource, as forumsys.com might go down or block repeated request from us if we start hammering them.

Comment (shellymiron) [reply tests/utils/constants/constants.go:36]: true, I added the /hold also for that purpose 
@xueli181114 - I see that in ocm-backend repo you are using `myldap.com` or `forumsys.com`
do you think this is a solution or do we have other option? creating new ldap sever seems to be complex

File: tests/utils/exec/idps.go [hunk 2]
```diff
@@ -9,12 +9,15 @@ import (
...
-	ClusterID     string        `json:"cluster_id,omitempty"`
-	Name          string        `json:"name,omitempty"`
-	ID            string        `json:"id,omitempty"`
-	Token         string        `json:"token,omitempty"`
-	OCMENV        string        `json:"ocm_environment,omitempty"`
-	URL           string        `json:"url,omitempty"`
+	ClusterID string `json:"cluster_id,omitempty"`
+	Name      string `json:"name,omitempty"`
+	ID        string `json:"id,omitempty"`
+	Token     string `json:"token,omitempty"`
```
Comment (amalykhi) [inline tests/utils/exec/idps.go:15]: Run "terraform fmt" , please

Comment (xueli181114) [reply tests/utils/exec/idps.go:15]: It's go code. You can install extension to keep code fmt alignment automatically.

Comment (amalykhi) [reply tests/utils/exec/idps.go:15]: which extension? for VS Code?

Comment (xueli181114) [reply tests/utils/exec/idps.go:15]: I have `Go Grammar` and "Go Outliner" installed.

**Comment by shellymiron** [comment]
@xueli181114 @yasun1 Hey! can you please help to review? :)

File: tests/utils/exec/idps.go [hunk 3]
```diff
@@ -15,8 +15,11 @@ type IDPArgs struct {
 	Token         string        `json:"token,omitempty"`
 	OCMENV        string        `json:"ocm_environment,omitempty"`
 	URL           string        `json:"url,omitempty"`
+	CA            string        `json:"ca,omitempty"`
+	Insecure      bool          `json:"insecure,omitempty"`
 	MappingMethod string        `json:"mapping_method,omitempty"`
 	HtpasswdUsers []interface{} `json:"htpasswd_users,omitempty"`
+	// Attributes []string      `json:"attributes,omitempty"`
```
Comment (amalykhi) [inline tests/utils/exec/idps.go:22]: Is it not relevant parameter? delete?

Comment (shellymiron) [reply tests/utils/exec/idps.go:22]: It is not a mandatory field in order for the test to work, but I was thinking maybe in future tests
we will need to validate also all the optional fields for an idp. but anyway I'll delete it for this PR

**Pull request merged**